apps/web/src/app/tracks/_components/enroll-button.tsx

```diff
@@ -28,11 +33,12 @@ export function ActionButton({ action, text, trackId, slug }: EnrollButtonProps)
             variant: 'success',
             description: <p>You're now successfully {text.toLowerCase()}ed the track.</p>,
           });
-          setIsLoading(false);
-        } catch (error) {
-          error instanceof Error && error.message === 'User is not logged in'
-            ? toast({
-                title: `Please Login to continue`,
+        } catch (error: unknown) { // Here we use `unknown` to dealwith unknow errors
+          if (error instanceof Error) {
+            const customError = error as CustomError;
+            if (customError.message === 'User is not logged in' || customError.statusCode === 401) {
+              toast({
+                title: 'Please Login to continue',
```

Comment:
Can we add a redirect back to the page they are on, I believe you can just add a queryparam. check the codebase for examples

Reply:
Okay, I'll do this as well.